server/src/main/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapper.java

```diff
@@ -144,7 +144,9 @@ public void postParse(DocumentParserContext context) throws IOException {
         assert fieldType().isIndexed() == false;
 
         TimeSeriesIdBuilder timeSeriesIdBuilder = (TimeSeriesIdBuilder) context.getDimensions();
-        context.doc().add(new SortedDocValuesField(fieldType().name(), timeSeriesIdBuilder.build().toBytesRef()));
+        BytesRef timeSeriesId = timeSeriesIdBuilder.build().toBytesRef();
+        context.doc().add(new SortedDocValuesField(fieldType().name(), timeSeriesId));
+        TimeSeriesModeIdFieldMapper.INSTANCE.createField(context, timeSeriesId);
```

Comment:
We've talked about this a few times. All of the options are kind of bad. I'll save this for a follow up change because it'll be a rename. And, we could spend a long time trying to find the right names....